mart/attack/adversary.py

```diff
@@ -151,6 +151,8 @@ def configure_gradient_clipping(
             for group in optimizer.param_groups:
                 self.gradient_modifier(group["params"])
 
+    # Turn off the inference mode, so we will create perturbation that requires gradient.
+    @torch.inference_mode(False)
```

Comment:
Why is this necessary now? I thought PL manages this already?

Reply:
Anomalib turns on the inference mode as we run `anomalib test`.

MART's trainer turns off the inference mode by default, as in https://github.com/IntelLabs/MART/blob/05886f7606872f69480eebdb8ee0435c8b873b48/mart/configs/trainer/default.yaml#L19

But Anomalib has its own trainer.